File: app/src/main/java/com/jonasgerdes/stoppelmap/model/shared/RealmString.java
package com.jonasgerdes.stoppelmap.model.shared;

import io.realm.RealmObject;

/**
 * Created by Jonas on 19.07.2016.
 */
public class RealmString extends RealmObject {
    private String val;

    public RealmString() {
    }

    public RealmString(String val) {
        this.val = val;
    }

    public String getVal() {
        return val;
    }

    public void setVal(String val) {
        this.val = val;
    }
}

File: app/src/main/java/com/jonasgerdes/stoppelmap/model/realm_wrapper/RealmString.kt
package com.jonasgerdes.stoppelmap.model.realm_wrapper

import com.google.gson.stream.JsonReader
import com.google.gson.stream.JsonWriter
import io.realm.RealmObject

/**
 * @author Jonas Gerdes <dev@jonasgerdes.com>
 * @since 15.06.2017
 */
open class RealmString(var value: String? = null) : RealmObject() {

    class TypeAdapter : com.google.gson.TypeAdapter<RealmString>() {
        override fun write(out: JsonWriter?, value: RealmString?) {
            //ignore
        }

        override fun read(reader: JsonReader?): RealmString {
            return RealmString(reader?.nextString())
        }

    }
}